packages/js/uri-resolvers/src/cache/WrapperCacheResolver.ts

```diff
@@ -82,55 +74,36 @@ export class PackageToWrapperCacheResolver<TError>
       resolutionContext.trackStep({
```

Comment:
IMO, it is the responsibility of the resolver to not only apply resolution logic, but to build a resolution context matching that resolution logic, I'm not sure we can separate it in most cases.
That said, if you have a good example how this can work in a base class (for all use cases) then we can create an issue for that to be solved in another PR

Reply:
Okay I can understand that perspective, but I'm still not 100% clear on what the different cases we want to support here. I was thinking they'd all be the same, but I can understand if that's not the case.